src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -386,6 +560,10 @@ pub fn try_drop_invalid_columns(
         }
     }
 
-    let row_bytes = Serializer::new(&column_ids).serialize_raw(datums);
+    // Data types are only needed when we are actually serializing. But we have encoded data here.
+    // Simple pass dummy data types.
+    let dummy_data_types = vec![DataType::Boolean; column_ids.len()];
+
+    let row_bytes = Serializer::new(&column_ids, dummy_data_types).serialize_raw(datums);
```

Comment:
Not a big deal but perhaps we need a `RawSerializer` (or just a bare function) rather than `Serializer`